be/src/vec/functions/function_struct_in.h

```diff
@@ -62,6 +62,11 @@ struct ColumnRowRef {
     }
 };
 
+struct StructInState {
+    std::unordered_set<ColumnRowRef, ColumnRowRef> args_set;
```

Comment:
use class and add ENABLE_FACTORY_CREATOR

Reply:
use ENABLE_FACTORY_CREATOR done!